### PR TITLE
Snj 27 qa bug fixes

### PR DESCRIPTION
**Pull request by anilasimbilen**

### CODE REVIEW

**Comment by anilasimbilen** [comment]
@michaelavoyan

**Review by michaelavoyan — CHANGES_REQUESTED** [review]

**Review by michaelavoyan — APPROVED** [review]

File: packages/sdk/src/impl/data/infrastructure/jwt/JwtServiceImpl.ts [hunk 1]
```diff
@@ -48,24 +39,36 @@ export default class JwtServiceImpl implements JwtService {
...
-        new SignJWT(jwtDescriptor.payload)
+        let signedJwtRes = new SignJWT(jwtDescriptor.payload)
             .setProtectedHeader(header)
             .setAudience(jwtDescriptor.iss)
             .setIssuer(jwtDescriptor.iss)
             .setJti(jwtDescriptor.jti)
-            .setIssuedAt(Date.now())
-            .setNotBefore(Date.now())
-            .setExpirationTime(new Date().addDaysToNow(7).getTime())
+            .setIssuedAt(undefined)
```
Comment (michaelavoyan) [inline packages/sdk/src/impl/data/infrastructure/jwt/JwtServiceImpl.ts:47]: `setIssuedAt(undefined)`
`setNotBefore(0)`
both values should be current time - see Android implementation.
@anilasimbilen

File: packages/sdk/src/impl/data/repositories/CountriesRepositoryImpl.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        return this.fetchCountries(Urls.Countries);
+    }
+
+    private async fetchCountries(
+        endpoint: string
+    ): Promise<VCLResult<VCLCountries>> {
+        let result = await this.networkService.sendRequest({
+            endpoint,
+            contentType: Request.ContentTypeApplicationJson,
+            method: "GET",
```
Comment (michaelavoyan) [inline packages/sdk/src/impl/data/repositories/CountriesRepositoryImpl.ts:24]: Please define constants for "GET" and "POST"
@anilasimbilen

File: packages/sdk/src/impl/data/repositories/CredentialTypeSchemaRepositoryImpl.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            Urls.CredentialTypeSchemas + schemaName
+        );
+    }
+
+    private async fetchCredentialTypeSchema(
+        endpoint: string
+    ): Promise<VCLResult<VCLCredentialTypeSchema>> {
+        let result = await this.networkService.sendRequest({
+            endpoint,
+            method: "GET",
```
Comment (michaelavoyan) [inline packages/sdk/src/impl/data/repositories/CredentialTypeSchemaRepositoryImpl.ts:25]: same const for "GET"

File: packages/sdk/src/impl/data/repositories/CredentialTypesRepositoryImpl.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        return await this.fetchCredentialTypes(endpoint);
+    }
+
+    async fetchCredentialTypes(
+        endpoint: string
+    ): Promise<VCLResult<VCLCredentialTypes>> {
+        let result = await this.networkService.sendRequest({
+            endpoint,
+            contentType: Request.ContentTypeApplicationJson,
+            method: "GET",
```
Comment (michaelavoyan) [inline packages/sdk/src/impl/data/repositories/CredentialTypesRepositoryImpl.ts:26]: same

File: packages/sdk/src/impl/data/repositories/ExchangeProgressRepositoryImpl.ts [hunk 5]
```diff
@@ -11,11 +11,10 @@ export default class ExchangeProgressRepositoryImpl
...
-        exchangeDescriptor: VCLExchangeDescriptor,
-        completionBlock: (r: VCLResult<VCLExchange>) => any
-    ): void {
-        this.networkService.sendRequestRaw({
+    async getExchangeProgress(
+        exchangeDescriptor: VCLExchangeDescriptor
+    ): Promise<VCLResult<VCLExchange>> {
+        let submissionResult = await this.networkService.sendRequest({
             useCaches: false,
             method: "GET",
```
Comment (michaelavoyan) [inline packages/sdk/src/impl/data/repositories/ExchangeProgressRepositoryImpl.ts:19]: same

File: packages/sdk/src/impl/data/repositories/GenerateOffersRepositoryImpl.ts [hunk 6]
```diff
@@ -29,27 +28,20 @@ export default class GenerateOffersRepositoryImpl
             body: generateOffersDescriptor.payload,
             method: "POST",
```
Comment (michaelavoyan) [inline packages/sdk/src/impl/data/repositories/GenerateOffersRepositoryImpl.ts:29]: same

File: packages/sdk/src/impl/data/repositories/SubmissionRepositoryImpl.ts [hunk 7]
```diff
@@ -10,12 +11,11 @@ import { HeaderKeys, HeaderValues } from "./Urls";
...
-        jwt: VCLJwt,
-        completionBlock: (r: VCLResult<VCLSubmissionResult>) => any
-    ): void {
-        this.networkService.sendRequestRaw({
+        jwt: VCLJwt
+    ): Promise<VCLResult<VCLSubmissionResult>> {
+        let result = await this.networkService.sendRequest({
             endpoint: submission.submitUri,
             body: submission.generateRequestBody(jwt),
             method: "POST",
```
Comment (michaelavoyan) [inline packages/sdk/src/impl/data/repositories/SubmissionRepositoryImpl.ts:23]: same

File: packages/sdk/src/impl/data/repositories/VerifiedProfileRepositoryImpl.ts [hunk 8]
```diff
@@ -10,11 +10,10 @@ export default class VerifiedProfileRepositoryImpl
...
-    getVerifiedProfile(
-        verifiedProfileDescriptor: VCLVerifiedProfileDescriptor,
-        completionBlock: (r: VCLResult<VCLVerifiedProfile>) => any
-    ): void {
-        this.networkService.sendRequestRaw({
+    async getVerifiedProfile(
+        verifiedProfileDescriptor: VCLVerifiedProfileDescriptor
+    ): Promise<VCLResult<VCLVerifiedProfile>> {
+        let result = await this.networkService.sendRequest({
             method: "GET",
```
Comment (michaelavoyan) [inline packages/sdk/src/impl/data/repositories/VerifiedProfileRepositoryImpl.ts:17]: same

**Review by michaelavoyan — CHANGES_REQUESTED** [review]

File: packages/sdk/src/api/VCL.ts [hunk 9]
```diff
@@ -95,28 +77,14 @@ export default interface VCL {
...
-    ): void;
+    generateSignedJwt(jwtDescriptor: VCLJwtDescriptor): Promise<VCLJwt>;
 
-    generateDidJwk(
-        successHandler: (jwk: VCLDidJwk) => any,
-        errorHandler: (e: VCLError) => any
-    ): void;
+    generateDidJwk(): Promise<VCLDidJwk>;
 
     printVersion(): void;
```
Comment (michaelavoyan) [inline packages/sdk/src/api/VCL.ts:89]: `printVersion()` shouldn't be part of the public API.

**Review by michaelavoyan — CHANGES_REQUESTED** [review]

File: packages/sdk/src/impl/data/infrastructure/jwt/JwtServiceImpl.ts (hunk 1 above)
Comment (anilasimbilen) [reply packages/sdk/src/impl/data/infrastructure/jwt/JwtServiceImpl.ts:47]: Yes, the default value for issuedAt is current time at jose library.

**Review by michaelavoyan — APPROVED** [review]

**Pull request merged**